controller/invitation.go

```diff
@@ -126,25 +128,37 @@ func (c *InvitationController) AcceptInvite(ctx *app.AcceptInviteInvitationConte
 		}, "failed to accept invitation, invalid code")
 
 		errResponse := err.Error()
-		redirectURL, err = rest.AddParam(redirectURL, "error", errResponse)
-		ctx.ResponseData.Header().Set("Location", redirectURL)
+		acceptedRedirectURL, err = rest.AddParam(acceptedRedirectURL, "error", errResponse)
+		ctx.ResponseData.Header().Set("Location", acceptedRedirectURL)
 		return ctx.TemporaryRedirect()
 	}
 
-	_, err = c.app.InvitationService().Accept(ctx, acceptCode)
+	_, absoluteSpacePath, err := c.app.InvitationService().Accept(ctx, acceptCode)
 
 	if err != nil {
 		errResponse := err.Error()
-		redirectURL, err = rest.AddParam(redirectURL, "error", errResponse)
+		acceptedRedirectURL, err = rest.AddParam(acceptedRedirectURL, "error", errResponse)
 		if err != nil {
 			return err
 		}
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to accept invitation")
+
+		ctx.ResponseData.Header().Set("Location", acceptedRedirectURL)
+		return ctx.TemporaryRedirect()
 	}
 
+	redirectURL := generateInvitationAcceptedRedirectURL(c.config.GetUIURL(), absoluteSpacePath)
+
 	log.Debug(ctx, map[string]interface{}{
 		"accept-code": ctx.AcceptCode,
 	}, "invitation accepted")
 
 	ctx.ResponseData.Header().Set("Location", redirectURL)
 	return ctx.TemporaryRedirect()
 }
+
+func generateInvitationAcceptedRedirectURL(uiURL, path string) string {
+	return fmt.Sprintf("%s/%s", uiURL, path)
```

Comment:
not sure you really need this function, it's a 1-liner (unless you plan to call a remote service in the future?)